### ISSUE TITLE
[REQUEST] docker support

### ISSUE THREAD

**Issue by szabogyula** [aristocratos/btop#256]
Hi,

Is there any plan to support monitor docker containers images and other docker things similar to https://github.com/jesseduffield/lazydocker?

It should be great.

Thanks

**Comment by pedro-cf** [comment]
+1 for Docker support. I'd like to expand on this feature request with my daily use case and detailed requirements:

I work heavily with Docker and find myself constantly switching between btop++ for system monitoring and other tools (like lazydocker and docker stats) for container stats. While btop++ is my favorite system monitor, I have to juggle multiple terminals just to keep an eye on my containers. It breaks my workflow and feels clunky.

What I envision for Docker support in btop++:
- Container stats seamlessly integrated with system metrics
- Process grouping by containers (essential for microservices)
- Key container info: health status, uptime, image, port mappings
- All while keeping btop++'s fantastic interface

I've tried various alternatives:
- ctop (too container-focused, missing system context)
- docker stats (basic CLI output only)
- lazydocker (great but requires separate window)
- Portainer (overkill for quick monitoring)

None provide that unified view that btop++ could offer by integrating container metrics into its existing, polished UI.

The feature could be optional (like other btop++ features) - enabled via configuration for those who need it. For those of us working with containers daily, it would eliminate the need to constantly switch between monitoring tools.

Would be happy to provide more specific use cases or testing if this moves forward!

**Comment by tkoop** [comment]
"glances" has Docker support. I think btop should too.

**Comment by MaddyGuthridge** [comment]
In particular, I would want to know the container of processes on my server. This would make it much easier to identify where different processes come from, so that I can more-easily identify when a process is misbehaving.

<img width="529" height="690" alt="Image" src="https://github.com/user-attachments/assets/5e6aa886-2518-48bf-9f5d-814b248506ba" />

Not to mention that I will feel far less anxious if I can tell that all those random node servers running as root are in containers.

**Comment by pomarec** [comment]
I hope its not out of topic.
I'm pretty much in the same situation as @pedro-cf.
I also have various [docker contexts](https://docs.docker.com/engine/manage-resources/contexts/) (essentialy VMs with a docker engine installed).

I would love to be able to run btop and then be able to choose a context somewhere in the interface. This would switch to the stats of the selected context's underlying vm.

**Comment by diktomat** [comment]
Adding to Docker, it would be great for btop to display other kinds of containers too, like FreeBSD jails, or Linux's LXC and systemd-nspawn. It could just be adding @<container-name> to the user column, with a filter option being the cherry on top.

OT: Besides, @MaddyGuthridge, *nothing* should run as root unless strictly necessary, not even inside containers!

**Comment by MaddyGuthridge** [comment]
@diktomat agreed, unfortuntely it is not my docker image being run (I have ensured all of my personal projects do not run as `root`). This is part of the reason why something like this feature would be helpful. I'd know where to go to open up a bug report.
